src/api/spec/queries/notifications_finder_spec.rb

```diff
@@ -0,0 +1,131 @@
+RSpec.describe NotificationsFinder do
```

Comment:
What are you testing here? `Hash.map`?

Reply:
The idea was to test that I didn't break anything by refactoring. That's how it's done, right? Test, refactor.

Reply:
And how will this not break if you refactor and introduce the test together? And how will this break in the future?

Reply:
The first commit is the test, passing in green. Then the refactor, that keeps passing in green. Successfully refactored!
Another thing is: do we need that test in the CI? No, once that refactor is done, that test is useless, but I left it there, because:
1) Was not sure about the whole PR
2) Prove to the reviewers that the refactoring didn't break anything